### DISCUSSION TITLE
Change color with OnPlatform using DynamicResource

### DISCUSSION THREAD

**Discussion by StefanoRivolta-Previero** [AvaloniaUI/Avalonia#13963] [Q&A] [answered]
Hi, starting to dip my toes in Avalonia, and I'm having a couple issues with the cross-platform styling part.
I'm trying to change the PlatformInfoPage from the ControlCatalog to use a Resource instead of just Green, but I'm not having any luck. Am I missing something?

Original:
```
<Border Height="100" Width="100" Background="{OnPlatform Gray, Windows=Green}">
  <TextBlock Text="Windows" />
</Border>
```

What I've tried but doesn't work:
```
<Border Height="100" Width="100" Background="{OnPlatform Gray, Windows={DynamicResource CatalogBaseMediumColor}}">
  <TextBlock Text="Windows" />
</Border>
```

Thank you for the help. 
Btw, coming from WPF I'm having a blast with Avalonia, great work!

**Comment by timunie** [comment, marked as answer]
`DynamicResource` is not supported here, only `StaticResource`. If you need DynmaicResouces, you will need to write a custom converter or MarupExt.

**Reply by StefanoRivolta-Previero** [reply]
Thank you very much. 
Btw, I've found this weird behavior, essentially this code works as expected:

`Background="{OnPlatform {StaticResource CatalogBaseMediumColor}, Android={StaticResource GreenColor}}"`

While this code give a `System.InvalidCastException: 'Specified cast is not valid.'` when run

`Background="{OnPlatform Gray, Android={StaticResource GreenColor}}"`

It does this with any color resource. It does that also with SolidColorBrush resource.

**Reply by maxkatz6** [reply]
DynamicBindings should work even now. 
but mixing different types is going to be a bad idea, as compiler has a hard time guessing what type to use for a value.

**Reply by maxkatz6** [reply]
StaticResource returns an object
DynamicResource returns a IBinding
And “Gray” is just a string, that could be converted to a Color compile time, if compiler had enough information. But having “object” in a second branch breaks predictions.
